### PR TITLE
Feature #5764. SQL: Implement Trigonometric Functions

### PR DESCRIPTION
**Pull request by Sadredneck**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+import javax.ws.rs.core.Response;
+
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+
+public class SqlTrigonometricFunctionTest extends SqlTest {
+
+    @DataProvider
+    public static Object[][] provideSinParameters() {
+        return new Object[][]{{"-pi()", "-1.2246467991473532E-16"}, {"-pi()*3/4", "-0.7071067811865476"},
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java]: Where do you get the expected values from?

Comment (Sadredneck) [reply src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java]: Select sin(-pi())

Comment (raipc) [reply src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java]: You must not trust the system under test. Check that the result does not differ from calculated value more than some small epsilon

Comment (Sadredneck) [reply src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java]: Expected values takes from Math functions and asserts with delta

File: src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                {"pi()/2", Math.PI / 2},
+                {"pi()*3/4", Math.PI * 3 / 4},
+                {"pi()", Math.PI}};
+    }
+
+    private static Object[][] getInverseArguments() {
+        return new Object[][]{{"-1", -1.0}, {"-1/2", -1.0 / 2.0}, {"0", 0.0}, {"1/2", 1.0 / 2.0}, {"1", 1.0}};
+    }
+
+    private static Object[][] getTrigonometricData(TrigonometricAccessor[] accessors, Object[][] arguments) {
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java]: simplify

File: src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            this.functionName = functionName;
+            this.trigonometricFunction = trigonometricFunction;
+        }
+
+        double apply(double parameter) {
+            return trigonometricFunction.apply(parameter);
+        }
+    }
+
+    private static Object[][] getArguments() {
```
[the remark this answers is not in the document]
Comment (Sadredneck) [reply src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java]: Renamed

File: src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java (hunk 2 above)
Comment (Sadredneck) [reply src/test/java/com/axibase/tsd/api/method/sql/function/math/SqlTrigonometricFunctionTest.java]: Changed method

**Pull request merged**